src/main.js

```diff
@@ -64,29 +56,35 @@ const filmsListTemplate = filmsTemplate.querySelector('.films-list');
 
 
 const filmsListContainerTemplate = filmsListTemplate.querySelector('.films-list__container');
-renderFilmCards(minFilms, filmsListContainerTemplate);
+renderFilmCards(filmsListContainerTemplate);
+
 
-render(filmsListTemplate, createButtonShowMoreTemplate(), 'beforeend');
 render(filmsTemplate, createFilmListTopRatedTemplate(), 'beforeend');
 render(filmsTemplate, createFilmListExtraTemplate(), 'beforeend');
 
-if (allFilms.length > FILM_CARDS_PER_STEP) {
-  let renderedFilmCount = FILM_CARDS_PER_STEP;
-
-  const showMoreButton = filmsListTemplate.querySelector('.films-list__show-more');
-  showMoreButton.addEventListener('click', (evt) => {
-    evt.preventDefault();
-    allFilms
-      .slice(renderedFilmCount, renderedFilmCount + FILM_CARDS_PER_STEP)
-      .forEach((film) => render(filmsListContainerTemplate, createFilmCardTemplate(film), 'beforeend'));
-
-    renderedFilmCount += FILM_CARDS_PER_STEP;
+const createCard = () => {
+  if (allFilms.length > FILM_CARDS_PER_STEP) {
+    render(filmsListTemplate, createButtonShowMoreTemplate(), 'beforeend');
+    let renderedFilmCount = FILM_CARDS_PER_STEP;
+
+    const showMoreButton = filmsListTemplate.querySelector('.films-list__show-more');
+    const createClickHandler = (evt) => {
+      evt.preventDefault();
+      allFilms
+        .slice(renderedFilmCount, renderedFilmCount + FILM_CARDS_PER_STEP)
+        .forEach((film) => render(filmsListContainerTemplate, createFilmCardTemplate(film), 'beforeend'));
+
+      renderedFilmCount += FILM_CARDS_PER_STEP;
+
+      if(renderedFilmCount >= allFilms.length) {
+        showMoreButton.remove();
+      }
+    };
+    showMoreButton.addEventListener('click', createClickHandler);
+  }
+};
 
-    if(renderedFilmCount >= allFilms.length) {
-      showMoreButton.remove();
-    }
-  });
-}
+createCard();
 
 const filmsListExtraTemplate = filmsTemplate.querySelectorAll('.films-list--extra')[0];
```

Comment:
```suggestion
const [filmsListExtraTemplate] = filmsTemplate.querySelectorAll('.films-list--extra');
```

используй реструктуризацию.